src/Chapter2.hs

```diff
@@ -336,7 +336,11 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList a b l
+    | null l = []
+    | a < 0 || b < 0 || b < a = []
+    | b > (length l) - 1 = [l !! x | x <- [a..(length l) - 1]]
+    | otherwise = [l !! x | x <- [a..b]]
```

Comment:
`length` and `!!` are quite expensive functions to walk the list. Here it would be more efficient to use `take` and `drop` functions that are faster and also they are streaming, so you won't nee to check `b > length l` as this is not an issue. Also, `take` and `drop` are not partial, meaning they won't fail at the runtime, while `!!` is partial and can fail 🙂

Reply:
Thank you very much for the tip. When I wrote that one it felt like I was trying to replicate a for-loop in C. `take` and `drop` are much better ideas.